lib/driver.rb

```diff
@@ -0,0 +1,109 @@
+require_relative 'csv_record'
+require 'csv'
+require 'time'
+require 'awesome_print'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :trips
+    attr_accessor :status #Wave 3: overriding status in dispatch, need to make this attr accessor
+    
+    def initialize(
+      id:,
+      name:,
+      vin:,
+      status: :AVAILABLE,
+      trips: nil
+      )
+      super(id)
+
+      @name = name 
+      @vin = vin
+      @status = status.to_sym #need to convert this to symbol
+      @trips = trips || []
+    
+      #Raise argument for vin
+      if @vin.length != 17
+        raise ArgumentError.new("Invalid VIN number, needs to be 17 in length.")
```

Comment:
It's helpful to include the bad argument in the message for your `ArgumentError`:

```suggestion
        raise ArgumentError.new("Invalid VIN number: #{@vin}, needs to be 17 in length.")
```